pages/en/zkapps/how-to-write-a-zkapp-ui.mdx

```diff
@@ -26,23 +25,32 @@ This page is a work in progress.
 
 ## Adding your smart contract as a dependency of the UI
 
-<Alert kind="info">
-
-You can integrate your smart contract by publishing it to npm and adding it as a  
- dependency of your UI.
-
-</Alert>
-
-#### Publish to npm
+#### Setup
 
 The `index.ts` file is the entry point of your project. This file only imports all smart contract classes you want access to, and exports them. This pattern allows you to specify which smart contracts are available to import when consuming your project from npm within your UI.
 
 ```ts
-import { YourSmartContract } from './YourSmartContract.js';
+import { YourSmartContract } from "./YourSmartContract.js";
 
 export { YourSmartContract };
 ```
 
+#### Local development
+
+The `npm link` command allows to connect your smart contract to your UI during local development without having to publish it npm.
+
+1. **Link your smart contract:** To link your smart contract package, enter `npm link` on the command line at the root of your smart contract project directory.
+2. **Add the smart contract link to your UI project:** To add the smart contract to your UI, enter `npm link your-package-name` on the command line at the root of your UI project directory. `your-package-name` is the same as the `name` property in the`package.json` of the smart contract project.
+3. Import the linked smart contract package into your UI using: `import { YourSmartContract } from ‘your-package-name’;` , where YourSmartContract is the named export that you chose in your smart contract.
+
```

Comment:
@ymekuria Should we finish this section by mentioning that that they need to run `npm run build` in their smart contract in order to make any changes in their smart contract available in their UI project? I assume it is

Reply:
I added this.

Reply:
That's a good idea @jasongitmail . I'll add it.

Reply:
Looks great!